jpa/subsystem/src/main/java/org/jboss/as/jpa/service/PersistenceUnitServiceImpl.java

```diff
@@ -169,11 +175,21 @@ public Void run() {
                                         if (beanManagerInjector.getOptionalValue() != null) {
                                             proxyBeanManager = new ProxyBeanManager();
                                             proxyBeanManager.setDelegate(beanManagerInjector.getOptionalValue());
-                                            properties.getValue().put(CDI_BEAN_MANAGER, proxyBeanManager);
+                                            wrapperBeanManagerLifeCycle = persistenceProviderAdaptor.beanManagerLifeCycle(proxyBeanManager);
```

Comment:
I don't know how this is meant to work, but if phaseOnePUS returned a non-null value, should that be overwritten as happens here?